lib/services/debug/darwin-debugger-service.ts

```diff
@@ -0,0 +1,95 @@
+
+		// The grep must be ios_webki.
+		// The output will look like this:
+		// ios_webki 26903 blackdragon    3u  IPv4 0x5377cd482a2d40e3      0t0  TCP *:9221 (LISTEN)
+		// ios_webki 26903 blackdragon    5u  IPv4 0x5377cd4846656da3      0t0  TCP *:teamcoherence (LISTEN)
+		let lsofResult = shelljs.exec("lsof -i | grep ios_webki").output;
+		let portRegExp = /TCP \*:(\d+)/g;
+
+		let devToolsPortMatches = portRegExp.exec(lsofResult);
+
+		if (!devToolsPortMatches) {
+			this.$errors.failWithoutHelp("Devtools failed to run. Please run the command again and if the problem exists reinstall ios-webkit-debug-proxy and try again.");
+		}
+
+		let devToolsPort = devToolsPortMatches[1];
+
+		shelljs.exec(`open -a safari http://localhost:${devToolsPort}`);
+	}
+
+	public debugAndroidApplication(applicationId: string): IFuture<void> {
+		return (() => {
+			let deviceIdentifier: string;
+			this.$androidDeviceDiscovery.startLookingForDevices().wait();
+			let connectedDevices = this.$devicesService.getDevicesForPlatform("android");
+
+			// Use setTimeout because $androidDeviceDiscovery.startLookingForDevices() kills the process and the prompter does not wait for choice.
+			setTimeout(() => {
+				fiberBootstrap.run(() => {
+					if (connectedDevices.length > 1) {
+						let devicesNames: string[] = connectedDevices.map((device: Mobile.IDevice) => device.deviceInfo.displayName);
+						let selectedDeviceName = this.$prompter.promptForChoice("You have more than one Android devices connected to your computer. Please choose which one to use", devicesNames).wait();
+
+						deviceIdentifier = connectedDevices.filter((device: Mobile.IDevice) => device.deviceInfo.displayName === selectedDeviceName)[0].deviceInfo.identifier;
+					} else if (connectedDevices.length === 1) {
+						deviceIdentifier = connectedDevices[0].deviceInfo.identifier;
+					} else {
+						deviceIdentifier = this.$androidEmulatorServices.startEmulator().wait();
+					}
+
+					let applicationsAvailableForDebugging = this.$androidProcessService.getApplicationsAvailableForDebugging(deviceIdentifier).wait();
+					let applicationNotStartedErrorMessage = `Application with identifier ${applicationId} is not started on device ${deviceIdentifier}. Please open the application on the device to debug it.`;
+
+					if (!_.contains(applicationsAvailableForDebugging, applicationId)) {
+						this.$errors.failWithoutHelp(applicationNotStartedErrorMessage);
+					}
+
+					let tcpPort: string;
+
+					try {
+						tcpPort = this.$androidProcessService.mapAbstractToTcpPort(deviceIdentifier, applicationId).wait();
+					} catch (err) {
+						console.log(err);
+						this.$errors.failWithoutHelp("Your device has no open ports. Please close programs that are using device's ports to listen on them and try again.");
+					}
+
+					// TODO: Find a way to start chrome with chrome://inspect:port.
+					shelljs.exec(`open -a "Google Chrome" http://localhost:${tcpPort}`);
```

Comment:
is Google Chrome the correct name on Mac. If I remember correctly, we have a variable in config.json that points to the real name and the users can change it on their machines